src/app/cli/src/cli_parser.rs

```diff
@@ -106,6 +107,10 @@ pub fn cli() -> Command {
                             .action(ArgAction::Append)
                             .index(1)
                             .help("Dataset manifest reference(s) (path, or URL)"),
+                        cli_arguments::add::PUBLIC.to_clap_arg_long()
+                            .hide(true)
+                            .action(ArgAction::SetTrue)
```

Comment:
How about using `--visibility public` instead of a flag?

If we ever add more kinds of visibility (`external` / `public:rw`?)  - it would allow easily extending without breaking the interface.